Support Android 29 in cross-compilation

matthewbauer opened the pull request:
Some miscellaneous commits for getting Android 29 to work. Google Play requires Android 29 for new apps starting August 3, 2020 and updates starting November 2, 2020.

https://developer.android.com/distribute/play-policies#apps-must-target-api-level-29

matthewbauer committed:
ghc: set Stage1Only = YES on iOS

We want stage1Only here even if system == system, since we can’t run
the native arch simulator binaries locally.

matthewbauer committed:
android: update sdk, ndk to 29, 21

matthewbauer committed:
androidenv: remove duplicate addons.nix entries

cdepillabout commented on pkgs/development/compilers/ghc/8.8.4.nix:
```diff
@@ -63,7 +63,7 @@ let
     INTEGER_LIBRARY = ${if enableIntegerSimple then "integer-simple" else "integer-gmp"}
   '' + stdenv.lib.optionalString (targetPlatform != hostPlatform) ''
-    Stage1Only = ${if targetPlatform.system == hostPlatform.system then "NO" else "YES"}
+    Stage1Only = ${if (targetPlatform.system == hostPlatform.system && !targetPlatform.isiOS) then "NO" else "YES"}
```
Would you be able to put a comment on this line (and in the other files)?  It is not immediately obvious to me why we need this `if` statement here.

matthewbauer committed:
android: readd 18b ndk

This is referenced in all-packages.nix, so might as well leave it
heref or now.

matthewbauer committed:
ghc: document Stage1Only setting
